src/main/java/spireMapOverhaul/zones/ambushed/AmbushedZone.java

```diff
@@ -0,0 +1,385 @@
+        return cardDrawCardIDs.stream()
+                .filter(id -> isCardForShop(id, rarity, type, isColored))
+                .filter(id -> !selectedCardIDs.contains(id)) // Exclude already selected card IDs
+                .collect(Collectors.toList());
+    }
+
+
+    private boolean isCardForShop(String cardID, AbstractCard.CardRarity rarity, AbstractCard.CardType type, boolean isColored) {
+        AbstractCard card = CardLibrary.getCard(cardID);
+        boolean isCardColored = card.color != AbstractCard.CardColor.COLORLESS;
+        return card != null && card.rarity == rarity && card.type == type && isCardColored == isColored;
+    }
+
+    @Override
+    public void postCreateShopPotions(ShopScreen screen, ArrayList<StorePotion> potions) {
+        for (StorePotion storePotion : potions) {
+            AbstractPotion replacementPotion;
+            switch (storePotion.potion.rarity) {
+                case COMMON:
+                    replacementPotion = PotionHelper.getPotion(SwiftPotion.POTION_ID);
+                    break;
+                case UNCOMMON:
+                    replacementPotion = PotionHelper.getPotion(GamblersBrew.POTION_ID);
+                    break;
+                case RARE:
+                    // Randomly select between Smoke Bomb and Snecko Oil for Rare potions
+                    if (AbstractDungeon.potionRng.randomBoolean()) {
+                        replacementPotion = PotionHelper.getPotion(SmokeBomb.POTION_ID);
+                    } else {
+                        replacementPotion = PotionHelper.getPotion(SneckoOil.POTION_ID);
+                    }
+                    break;
+                default:
+                    continue; // Skip if the potion has an unexpected rarity
+            }
+            storePotion.potion = replacementPotion;
+        }
+    }
+
+    @Override
+    public void modifyReward(RewardItem rewardItem) {
+        if (rewardItem.type == RewardItem.RewardType.POTION && rewardItem.potion != null) {
+            // Store the rarity before clearing the existing potion
+            AbstractPotion.PotionRarity potionRarity = rewardItem.potion.rarity;
+
+            // Clearing the existing potion
+            rewardItem.potion = null;
+
+            AbstractPotion replacementPotion;
+            switch (potionRarity) {
+                case COMMON:
+                    replacementPotion = PotionHelper.getPotion(SwiftPotion.POTION_ID);
+                    break;
+                case UNCOMMON:
+                    replacementPotion = PotionHelper.getPotion(GamblersBrew.POTION_ID);
+                    break;
+                case RARE:
+                    // Randomly select between Smoke Bomb and Snecko Oil for Rare potions
+                    if (AbstractDungeon.cardRandomRng.randomBoolean()) {
+                        replacementPotion = PotionHelper.getPotion(SmokeBomb.POTION_ID);
```

Comment:
Getting smoke bombs feels like a troll. Taking harder fights and having part of the "reward" be a near-useless potion isn't a great experience.

I'm not sure about the whole potion replacement thing either, but if you're doing it, it would land better if you cut the smoke bomb thing and just gave out Snecko Oils.